### PR TITLE
CORE-16879 - replace max retry attempts with max retry window duration

### PR DESCRIPTION
**Pull request by conal-smith**
Runtime-os change - https://github.com/corda/corda-runtime-os/pull/4587

### CODE REVIEW

File: data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
-          "description": "The maximum number of times Corda retries a transient error before failing the flow. A value of zero disables retries.",
+        "maxRetryWindowDuration": {
+          "description": "The duration in milliseconds after a transient error that Corda retries a flow before failing it. A value of zero disables retries.",
           "type": "integer",
           "minimum": 0,
-          "maximum": 1000,
-          "default": 5
+          "maximum": 2147483647,
+          "default": 120000
         },
```
Comment (jujoramos) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json]: Isn't this considered an API breaking change, as the flow configuration itself is modifiable by clients?.

Comment (JamesHR3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json]: Yeah it is - this should be in a new version of the schema.

Comment (sprintengine) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json]: Yes, unless we add maxRetryWindowDuration in a non-breaking way (as an optional config, leaving in maxRetryAttempts, even though it's non-functional). 

I'll pull this change out into a new version of the corda.flow.json, I assume the new config will be version 1.1

Comment (JamesHR3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json]: Thanks Conal - yes I think that's the right thing to do.

Comment (sprintengine) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json]: Looping in @simon-johnson-r3 for his opinion here, regarding https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4564091182/Making+a+Config+Change

File: data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json [hunk 2]
```diff
@@ -18,6 +18,13 @@
           "maximum": 1000,
           "default": 5
         },
+        "maxRetryWindowDuration": {
+          "description": "The duration in milliseconds after a transient error that Corda retries a flow before failing it. A value of zero disables retries.",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 2147483647,
+          "default": 120000
+        },
```
Comment (sprintengine) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json:27]: GET on /config/flow

```
{
    "section": "corda.flow",
    "sourceConfig": "",
    "configWithDefaults": "{\"event\":{\"maxRetries\":5,\"messageResendWindow\":60000},\"processing\":{\"cleanupTime\":600000,\"maxFlowSleepDuration\":900000,\"maxRetryAttempts\":5,\"maxRetryDelay\":16000,\"maxRetryWindowDuration\":120000},\"session\":{\"cleanupTime\":600000,\"p2pTTL\":300000,\"timeout\":1800000}}",
    "schemaVersion": {
        "major": 1,
        "minor": 0
    },
    "version": 0
}
```

Comment (sprintengine) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json:27]: POST to /config with
```
{
    "section": "corda.flow",
    "version": 0,
    "config": {
        "processing":{
            "maxRetryAttempts": 1,
            "maxRetryWindowDuration": 1
        }
    },
    "schemaVersion": {
        "major": 1,
        "minor": 0
    }
}
```

GET /config/flow:
```
{
    "section": "corda.flow",
    "sourceConfig": "{\"processing\":{\"maxRetryAttempts\":1,\"maxRetryWindowDuration\":1}}",
    "configWithDefaults": "{\"event\":{\"maxRetries\":5,\"messageResendWindow\":60000},\"processing\":{\"cleanupTime\":600000,\"maxFlowSleepDuration\":900000,\"maxRetryAttempts\":1,\"maxRetryDelay\":16000,\"maxRetryWindowDuration\":1},\"session\":{\"cleanupTime\":600000,\"p2pTTL\":300000,\"timeout\":1800000}}",
    "schemaVersion": {
        "major": 1,
        "minor": 0
    },
    "version": 1
}
```

File: data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json (hunk 1 above)
Comment (sprintengine) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json]: After discussing with Simon, we can't make breaking changes to config schema such as removing config properties due to reasons highlighted in the wiki page. I've added the new property with a default thus a non-breaking change.

**Review by jujoramos — DISMISSED** [review]

**Review by LWogan — APPROVED** [review]
LGTM

**Review by JamesHR3 — APPROVED** [review]

**Pull request merged**